app/scripts/services/paymentReportService.js

```diff
@@ -0,0 +1,95 @@
+angular
+  .module('confRegistrationWebApp')
+  .service('paymentReportService', function PaymentReportService(
+    $cacheFactory,
+    $rootScope,
+    $http,
+    $q,
+  ) {
+    const path = function(id) {
+      return 'conferences/' + (id || '') + '/payments/report';
```

Comment:
Is there any instance where we would generate this path and not have the id?

Reply:
It is always run in the context of conference. This path can be used for both viewing of existing reports and generation of new one for the conference based on currently existing payments

Reply:
I don't think you really answered the question. If the id can't be undefined, then you shouldn't need the `|| ''`. If it can be undefined, `'conferences//payments/report'` doesn't really make sense. Also you could use template literals.

Reply:
The or is removed now